### PR TITLE
upgrading to petitparser: ^4.3.0

### PR DESCRIPTION
**Pull request by dtylman**

### CODE REVIEW

File: lib/src/matcher.dart [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import 'package:query/src/ast.dart';
+
+//Matching Operators
+const String OP_LESS_OR_EQUAL = '<=';
```
Comment (isoos) [inline lib/src/matcher.dart:4]: Please follow Dart naming conventions, e.g. `lessOrEq` or `lessOrEqual`.

File: lib/src/matcher.dart [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+const String OP_LESS = '<';
+const String OP_MORE_OR_EQUAL = '>=';
+const String OP_MORE = '>';
+const String OP_NOT_EQAUL = '!=';
+const String OP_EQUAL = '=';
+
+abstract class QueryMatcher {
+  //AnyField indicates any field
+  static final String AnyField = '';
```
Comment (isoos) [inline lib/src/matcher.dart:13]: Do we need this here?

File: lib/src/matcher.dart [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  //getFieldMatcher returns a matcher for a specific field
+  QueryMatcher? getFieldMatcher(String field);
```
Comment (isoos) [inline lib/src/matcher.dart:15]: I'm not sure what this should do. It would be helpful to add a test that describes the various cases.

**Review by isoos — COMMENTED** [review]
Hey, thanks for this PR, but I'd need to get a bit more details worked out. If you want to, feel free to split it up and submit some parts (e.g. petitparser upgrade) as a separate PR.

**Comment by dtylman** [comment]
sorry for being so late.

I was a bit busy and its just went too old now....

**Pull request closed without merging**